Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts

```diff
@@ -98,11 +99,41 @@ export async function getNuGet(versionSpec: string, checkLatest?: boolean, addNu
     return fullNuGetPath;
 }
 
+// Based on code in Tasks\Common\MSBuildHelpers\msbuildhelpers.ts
+export async function getMSBuildVersionString(): Promise<string> {
+    let version: string;
+    const path: string = taskLib.which('msbuild', false);
+    if (path) {
+        const getVersionTool = taskLib.tool(path);
+        getVersionTool.arg(['/version', '/nologo']);
+        getVersionTool.on('stdout', (data: string) => {
+            if (data) {
+                version = data.toString().trim();
+                taskLib.debug('Found msbuild version: ' + version);
+            }
+        });
+        await getVersionTool.exec();
+    }
+    return version;
+}
+
+export async function getMSBuildVersion(): Promise<semver.SemVer> {
+    const version = await getMSBuildVersionString();
+    return semver.coerce(version);
```

Comment:
is this going to fail if version is null?

Reply:
passes the null through